app/components/chat/Chat.client.tsx

```diff
@@ -690,15 +731,29 @@ export const ChatImpl = memo(({ description, initialMessages, setInitialMessages
       return;
     }
 
-    webcontainer.then(async (wc) => {
-      await wc.fs.rm('/', { recursive: true, force: true });
-      await wc.fs.mkdir('/', { recursive: true });
+    // Show loading toast while retrying
+    const toastId = toast.loading('Loading previous version...');
 
-      const files = await fetchProjectFiles(repoStore.get().path, prevCommitHash);
-      await wc.mount(convertFileMapToFileSystemTree(files));
-      setMessages(messages.slice(0, messageIndex + 1));
-      reload();
-    });
+    try {
+      await webcontainer.then(async (wc) => {
```

Comment:
I think that we just do 

```typescript
const wc = await webcontainer;
await wc.fs.rm('/', { recursive: true, force: true });
// and so on
```

instead of `then()` chain